### PR TITLE
adds request sampler into zipkin tracing.

### LINKED ISSUE #812: Support RequestSampler in zipkin server middleware
`zipkin-go` [includes an option](https://github.com/openzipkin/zipkin-go/blob/master/middleware/http/server.go#L53) missing in this server middleware, `RequestSampler` which allows to change the sampling decision based on the request. This is specially useful when you want to sample requests coming from async operations (that often don't include tracing propagation) and in general to give more granularity to the sampling.

We are lacking that option in this server middleware, should I open a PR for that?

Ping @basvanbeek

### PR DESCRIPTION
**Pull request by jcchavezs**
Closes #812

Ping @basvanbeek

### CODE REVIEW

File: tracing/zipkin/http_test.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ const (
 	testTagValue = "test_value"
 )
 
-func TestHttpClientTracePropagatesParentSpan(t *testing.T) {
+func TestHTTPClientTracePropagatesParentSpan(t *testing.T) {
```
Comment (jcchavezs) [inline tracing/zipkin/http_test.go]: Http -> HTTP, not too important for this PR.

**Comment by jcchavezs** [comment]
Can we please merge this @basvanbeek @peterbourgon ?

File: tracing/zipkin/http.go [hunk 2]
```diff
@@ -156,6 +156,12 @@ func HTTPServerTrace(tracer *zipkin.Tracer, options ...TracerOption) kithttp.Ser
 
 			if config.propagate {
 				spanContext = tracer.Extract(b3.ExtractHTTP(req))
+
+				if config.requestSampler != nil && spanContext.Sampled == nil {
```
Comment (basvanbeek) [inline tracing/zipkin/http.go]: minor item... please swap the tested conditions.

Comment (jcchavezs) [reply tracing/zipkin/http.go]: Done!

**Pull request merged**